server/models/Profile.js

```diff
@@ -0,0 +1,45 @@
+const mongoose = require("mongoose");
+
+const profileSchema = new mongoose.Schema({
+  username: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: "user",
+  },

```

Comment:
This isn't a username except a `user` I would rename this property to be more descriptive.